corehq/apps/users/models.py

```diff
@@ -526,13 +526,24 @@ class Meta(object):
 
 class IsMemberOfMixin(DocumentSchema):
 
-    def _is_member_of(self, domain):
-        return domain in self.get_domains() or (
-            self.is_global_admin() and
-            not domain_restricts_superusers(domain)
-        )
+    def _is_member_of(self, domain, include_enterprise):
+        if self.is_global_admin() and not domain_restricts_superusers(domain):
+            return True
 
-    def is_member_of(self, domain_qs):
+        domains = self.get_domains()
+        if domain in domains:
+            return True
+
+        if include_enterprise:
+            for domain in domains:
+                master_link = get_domain_master_link(domain)
+                if master_link and not master_link.is_remote:
+                    if toggles.ENTERPRISE_LINKED_DOMAINS.enabled(master_link.master_domain):
+                        return self.get_domain_membership(master_link.master_domain)
```

Comment:
Is it possible to get here multiple domains would end up here, maybe if you were part of multiple enterprises? It seems like the value returned depends on the order of domains, since it early exits as soon as it gets here.

Reply:
Shouldn't have been looping in the first place. Fixed.